hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java

```diff
@@ -659,22 +659,27 @@ public boolean delete(Path f, boolean recursive) throws IOException {
         }
       }
 
+      // handling posix symlink delete behaviours
+      // i.) rm [-r] <symlink path>, delete symlink not target bucket contents
+      // ii.) rm -r <symlink path>/, delete target bucket contents not symlink
+      boolean handleTrailingSlash = false;
       boolean isBucketLink = false;
       // check for bucket link
       if (ofsPath.isBucket()) {
         isBucketLink = adapterImpl.getBucket(ofsPath, false)
             .isLink();
+        handleTrailingSlash = f.toString().endsWith(OZONE_URI_DELIMITER);
       }
 
       // if link, don't delete contents
-      if (isBucketLink) {
+      if (isBucketLink && !handleTrailingSlash) {
```

Comment:
Need to update the comment above accordingly

Reply:
Thanks @smengcl for your review comments and for following up the PR.  Latest commit to address review comments.

comment updated.